test/src/tezos.test.js

```diff
@@ -1,148 +1,51 @@
 import "core-js/stable";
 import "regenerator-runtime/runtime";
-import secp256k1 from "secp256k1";
-import sha256 from "js-sha256";
 import "./protocol_pb.js";
 import {
-    getSerializedAddressParameters,
-    numberToBigEndianBuffer,
-    swapTestPrivateKey,
-    partnerSerializedNameAndPubKey, DERSignatureOfPartnerNameAndPublicKey,
-    XTZConfig, XTZConfigSignature
-} from "./common"; import Exchange from "./exchange.js";
-import { TransportStatusError } from "@ledgerhq/errors";
-
-export const TEZOS_DERIVATION_PATH = "44'/1729'/0'/0'";
-export const TEZOS_PAYOUT_ADDRESS = "tz1RVYaHiobUKXMfJ47F7Rjxx5tu3LC35WSA";
-
-export const TEZOS_DERIVATION_PATH_2 = "44'/1729'/1'/0'";
-export const TEZOS_PAYOUT_ADDRESS_2 = "tz1RjJLvt7iguJQnVVWYca2AHDpHYmPJYz4d";
-
-export const TEZOS_INVALID_PAYOUT_ADDRESS = "tz1RVYaHiobUKXMfJ47F7Rjxx5tu3LC35WSB"; // notice the B at the end
-
-import { waitForAppScreen, zemu, nano_environments } from './test.fixture';
-
-test('[Nano S] Wrong XTZ payout address should not be accepted', zemu(nano_environments[0], async (sim) => {
-    const swap = new Exchange(sim.getTransport(), 0x00);
-    const transactionId: string = await swap.startNewTransaction();
-    await swap.setPartnerKey(partnerSerializedNameAndPubKey);
-    await swap.checkPartner(DERSignatureOfPartnerNameAndPublicKey);
-    var tr = new proto.ledger_swap.NewTransactionResponse();
-    tr.setPayinAddress("2324234324324234");
-    tr.setPayinExtraId("");
-    tr.setRefundAddress(TEZOS_INVALID_PAYOUT_ADDRESS);
-    tr.setRefundExtraId("");
-    tr.setPayoutAddress(TEZOS_INVALID_PAYOUT_ADDRESS);
-    tr.setPayoutExtraId("");
-    tr.setCurrencyFrom("BTC");
-    tr.setCurrencyTo("XTZ");
-    // 1 BTC to 10 XTZ
-    tr.setAmountToProvider(numberToBigEndianBuffer(100000000));
-    tr.setAmountToWallet(numberToBigEndianBuffer(1000000000));
-    tr.setDeviceTransactionId(transactionId);
-
-    const payload: Buffer = Buffer.from(tr.serializeBinary());
-    await swap.processTransaction(payload, 10000000);
-    const digest: Buffer = Buffer.from(sha256.sha256.array(payload));
-    const signature: Buffer = secp256k1.signatureExport(secp256k1.sign(digest, swapTestPrivateKey).signature);
-    await swap.checkTransactionSignature(signature);
-    const params = await getSerializedAddressParameters(TEZOS_DERIVATION_PATH);
-    await expect(swap.checkPayoutAddress(XTZConfig, XTZConfigSignature, params.addressParameters))
-        .rejects.toEqual(new TransportStatusError(0x6a83));
+    BTC_INFO,
+    XTZ_INFO,
+} from "./common";
+
+import { SwapTransactionPerformer } from "./SwapTransactionPerformer"
+
+import { zemu, nano_environments } from './test.fixture';
+
+test(`[Nano S] Tezos wrong payout address should not be accepted`, zemu(nano_environments[0], async (sim) => {
+    var test = new SwapTransactionPerformer(nano_environments[0], sim);
```

Comment:
Nits:
- `const` (better to forbid `var` completely imho; always `const`, or `let` if `const` doesn't work)
- the name `test` as a variable shadows the call to the `test` function from jest; maybe just `t` so that it's short? General I prefer descriptive name, but here minimizing the width of the rows might make sense.